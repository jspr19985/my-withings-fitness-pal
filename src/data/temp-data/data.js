export const stepsData = [
    { x: new Date("2022-10-19"), y: 0 },
    { x: new Date("2022-10-18"), y: 2195 },
    { x: new Date("2022-10-17"), y: 0 },
    { x: new Date("2022-10-16"), y: 8555 },
    { x: new Date("2022-10-15"), y: 8926 },
    { x: new Date("2022-10-14"), y: 0 },
    { x: new Date("2022-10-13"), y: 1053 },
    { x: new Date("2022-10-12"), y: 4475 },
    { x: new Date("2022-10-11"), y: 4734 },
    { x: new Date("2022-10-10"), y: 3524 },
    { x: new Date("2022-10-09"), y: 14554 },
    { x: new Date("2022-10-08"), y: 9981 }
];

export const weightData = [ // need some logic somewhere that fills in 0 with the day before (only for weight)
    { x: new Date("2022-10-19"), y: 179 },
    { x: new Date("2022-10-18"), y: 180 },
    { x: new Date("2022-10-17"), y: 181 },
    { x: new Date("2022-10-16"), y: 180 },
    { x: new Date("2022-10-15"), y: 179 },
    { x: new Date("2022-10-14"), y: 179 },
    { x: new Date("2022-10-13"), y: 180 },
    { x: new Date("2022-10-12"), y: 181 },
    { x: new Date("2022-10-11"), y: 181 },
    { x: new Date("2022-10-10"), y: 181 },
    { x: new Date("2022-10-09"), y: 180 },
    { x: new Date("2022-10-08"), y: 180 }
];

export const calorieData = [
    { x: new Date("2022-10-19"), y: 1207 },
    { x: new Date("2022-10-18"), y: 1533 },
    { x: new Date("2022-10-17"), y: 1326 },
    { x: new Date("2022-10-16"), y: 1368 },
    { x: new Date("2022-10-15"), y: 1611 },
    { x: new Date("2022-10-14"), y: 1157 },
    { x: new Date("2022-10-13"), y: 1076 },
    { x: new Date("2022-10-12"), y: 1379 },
    { x: new Date("2022-10-11"), y: 1278 },
    { x: new Date("2022-10-10"), y: 1169 },
    { x: new Date("2022-10-09"), y: 1299 },
    { x: new Date("2022-10-08"), y: 1124 }
];

export const bodyCompositionData = [
    [
        { date: new Date("2022-10-08"), comp: 51 },
        { date: new Date("2022-10-09"), comp: 51 },
        { date: new Date("2022-10-10"), comp: 58 },
        { date: new Date("2022-10-11"), comp: 52 },
        { date: new Date("2022-10-12"), comp: 51 },
        { date: new Date("2022-10-13"), comp: 52 },
        { date: new Date("2022-10-14"), comp: 51 },
        { date: new Date("2022-10-15"), comp: 52 },
        { date: new Date("2022-10-16"), comp: 52 },
        { date: new Date("2022-10-17"), comp: 52 },
        { date: new Date("2022-10-18"), comp: 53 },
        { date: new Date("2022-10-19"), comp: 52 }
    ],
    [
        { date: new Date("2022-10-08"), comp: 3 },
        { date: new Date("2022-10-09"), comp: 3 },
        { date: new Date("2022-10-10"), comp: 3 },
        { date: new Date("2022-10-11"), comp: 3 },
        { date: new Date("2022-10-12"), comp: 3 },
        { date: new Date("2022-10-13"), comp: 3 },
        { date: new Date("2022-10-14"), comp: 3 },
        { date: new Date("2022-10-15"), comp: 3 },
        { date: new Date("2022-10-16"), comp: 3 },
        { date: new Date("2022-10-17"), comp: 3 },
        { date: new Date("2022-10-18"), comp: 3 },
        { date: new Date("2022-10-19"), comp: 3 }
    ],
    [
        { date: new Date("2022-10-08"), comp: 46 },
        { date: new Date("2022-10-09"), comp: 46 },
        { date: new Date("2022-10-10"), comp: 39 },
        { date: new Date("2022-10-11"), comp: 45 },
        { date: new Date("2022-10-12"), comp: 46 },
        { date: new Date("2022-10-13"), comp: 45 },
        { date: new Date("2022-10-14"), comp: 46 },
        { date: new Date("2022-10-15"), comp: 45 },
        { date: new Date("2022-10-16"), comp: 45 },
        { date: new Date("2022-10-17"), comp: 45 },
        { date: new Date("2022-10-18"), comp: 44 },
        { date: new Date("2022-10-19"), comp: 45 }
    ]

];

export const macrosData = [
    [
        { date: new Date("2022-10-08"), macro: 64 },
        { date: new Date("2022-10-09"), macro: 99 },
        { date: new Date("2022-10-10"), macro: 74 },
        { date: new Date("2022-10-11"), macro: 83 },
        { date: new Date("2022-10-12"), macro: 89 },
        { date: new Date("2022-10-13"), macro: 52 },
        { date: new Date("2022-10-14"), macro: 62 },
        { date: new Date("2022-10-15"), macro: 68 },
        { date: new Date("2022-10-16"), macro: 62 },
        { date: new Date("2022-10-17"), macro: 56 },
        { date: new Date("2022-10-18"), macro: 69 },
        { date: new Date("2022-10-19"), macro: 50 }
    ],
    [
        { date: new Date("2022-10-08"), macro: 124 },
        { date: new Date("2022-10-09"), macro: 157 },
        { date: new Date("2022-10-10"), macro: 140 },
        { date: new Date("2022-10-11"), macro: 175 },
        { date: new Date("2022-10-12"), macro: 172 },
        { date: new Date("2022-10-13"), macro: 144 },
        { date: new Date("2022-10-14"), macro: 150 },
        { date: new Date("2022-10-15"), macro: 173 },
        { date: new Date("2022-10-16"), macro: 160 },
        { date: new Date("2022-10-17"), macro: 170 },
        { date: new Date("2022-10-18"), macro: 177 },
        { date: new Date("2022-10-19"), macro: 168 }
    ],
    [
        { date: new Date("2022-10-08"), macro: 48 },
        { date: new Date("2022-10-09"), macro: 53 },
        { date: new Date("2022-10-10"), macro: 49 },
        { date: new Date("2022-10-11"), macro: 30 },
        { date: new Date("2022-10-12"), macro: 59 },
        { date: new Date("2022-10-13"), macro: 39 },
        { date: new Date("2022-10-14"), macro: 22 },
        { date: new Date("2022-10-15"), macro: 40 },
        { date: new Date("2022-10-16"), macro: 42 },
        { date: new Date("2022-10-17"), macro: 25 },
        { date: new Date("2022-10-18"), macro: 28 },
        { date: new Date("2022-10-19"), macro: 28 }
    ]

];

export const exerciseData = [
    [
        { date: new Date("2022-10-08"), exercise: 0 },
        { date: new Date("2022-10-09"), exercise: 170 },
        { date: new Date("2022-10-10"), exercise: 0 },
        { date: new Date("2022-10-11"), exercise: 0 },
        { date: new Date("2022-10-12"), exercise: 0 },
        { date: new Date("2022-10-13"), exercise: 0 },
        { date: new Date("2022-10-14"), exercise: 0 },
        { date: new Date("2022-10-15"), exercise: 0 },
        { date: new Date("2022-10-16"), exercise: 0 },
        { date: new Date("2022-10-17"), exercise: 0 },
        { date: new Date("2022-10-18"), exercise: 0 },
        { date: new Date("2022-10-19"), exercise: 0 }
    ],
    [
        { date: new Date("2022-10-08"), exercise: 0 },
        { date: new Date("2022-10-09"), exercise: 380 },
        { date: new Date("2022-10-10"), exercise: 0 },
        { date: new Date("2022-10-11"), exercise: 0 },
        { date: new Date("2022-10-12"), exercise: 0 },
        { date: new Date("2022-10-13"), exercise: 0 },
        { date: new Date("2022-10-14"), exercise: 0 },
        { date: new Date("2022-10-15"), exercise: 0 },
        { date: new Date("2022-10-16"), exercise: 0 },
        { date: new Date("2022-10-17"), exercise: 0 },
        { date: new Date("2022-10-18"), exercise: 0 },
        { date: new Date("2022-10-19"), exercise: 0 }
    ],
    [
        { date: new Date("2022-10-08"), exercise: 400 },
        { date: new Date("2022-10-09"), exercise: 800 },
        { date: new Date("2022-10-10"), exercise: 800 },
        { date: new Date("2022-10-11"), exercise: 0 },
        { date: new Date("2022-10-12"), exercise: 0 },
        { date: new Date("2022-10-13"), exercise: 0 },
        { date: new Date("2022-10-14"), exercise: 0 },
        { date: new Date("2022-10-15"), exercise: 0 },
        { date: new Date("2022-10-16"), exercise: 0 },
        { date: new Date("2022-10-17"), exercise: 400 },
        { date: new Date("2022-10-18"), exercise: 400 },
        { date: new Date("2022-10-19"), exercise: 0 }
    ]

];
