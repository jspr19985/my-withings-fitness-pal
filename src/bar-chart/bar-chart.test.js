import BarChart from "./bar-chart";
import { render } from '@testing-library/react';
import { barData } from "../data/temp-data/data";

describe("App", () => {
    test('renders BarChart', () => {
        const { container } = render(<BarChart data={barData} />);
        expect(container).toMatchSnapshot();
    });
});
