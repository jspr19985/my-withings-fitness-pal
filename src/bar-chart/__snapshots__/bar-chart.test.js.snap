// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders BarChart 1`] = `
<div>
  <div
    class="VictoryContainer"
    style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
  >
    <svg
      height="150"
      role="img"
      style="pointer-events: all; width: 100%; height: 100%;"
      viewBox="0 0 450 150"
      width="450"
    >
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shape-rendering="auto"
          style="stroke: white; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
          vector-effect="non-scaling-stroke"
          x1="50"
          x2="400"
          y1="100"
          y2="100"
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="50"
            y1="100"
            y2="101"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-0-tickLabels-0"
            x="50"
            y="113.565"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="middle"
              x="50"
            >
              01 AM
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="400"
            x2="400"
            y1="100"
            y2="101"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-0-tickLabels-1"
            x="400"
            y="113.565"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="middle"
              x="400"
            >
              .001
            </tspan>
          </text>
        </g>
      </g>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shape-rendering="auto"
          style="stroke: white; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
          vector-effect="non-scaling-stroke"
          x1="50"
          x2="50"
          y1="50"
          y2="100"
        />
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="49"
            y1="93.58974358974359"
            y2="93.58974358974359"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-1-tickLabels-0"
            x="39"
            y="94.65474358974359"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="end"
              x="39"
            >
              0.2
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="49"
            y1="87.17948717948718"
            y2="87.17948717948718"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-1-tickLabels-1"
            x="39"
            y="88.24448717948718"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="end"
              x="39"
            >
              0.4
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="49"
            y1="80.76923076923077"
            y2="80.76923076923077"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-1-tickLabels-2"
            x="39"
            y="81.83423076923077"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="end"
              x="39"
            >
              0.6
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="49"
            y1="74.35897435897435"
            y2="74.35897435897435"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-1-tickLabels-3"
            x="39"
            y="75.42397435897435"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="end"
              x="39"
            >
              0.8
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="49"
            y1="67.94871794871796"
            y2="67.94871794871796"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-1-tickLabels-4"
            x="39"
            y="69.01371794871795"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="end"
              x="39"
            >
              1.0
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="49"
            y1="61.53846153846155"
            y2="61.53846153846155"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-1-tickLabels-5"
            x="39"
            y="62.603461538461545"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="end"
              x="39"
            >
              1.2
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: white; fill: transparent; size: 1px;"
            vector-effect="non-scaling-stroke"
            x1="50"
            x2="49"
            y1="55.12820512820513"
            y2="55.12820512820513"
          />
          <text
            direction="inherit"
            dx="0"
            id="chart-axis-1-tickLabels-6"
            x="39"
            y="56.19320512820513"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 3px; font-family: Major Mono Display; stroke-width: 0.2; stroke: white; letter-spacing: normal; padding: 10px; fill: #252525;"
              text-anchor="end"
              x="39"
            >
              1.4
            </tspan>
          </text>
        </g>
      </g>
      <g />
    </svg>
    <div
      style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
    >
      <svg
        height="150"
        style="overflow: visible; width: 100%; height: 100%;"
        viewBox="0 0 450 150"
        width="450"
      />
    </div>
  </div>
</div>
`;
